paddle/fluid/framework/details/reference_count_pass.cc

```diff
@@ -229,17 +302,43 @@ std::unique_ptr<ir::Graph> ReferenceCountPass::ApplyImpl(
         continue;
       }
 
-      bool ok;
-      auto result = ExtractComputationOpFromLastLivedVar(
-          name_var_pair.second.back(), i, shrink_func, &ok);
+      auto &var_name = name_var_pair.first;
+      auto &var_handles = name_var_pair.second;
+
+      for (auto iter = var_handles.rbegin(); iter != var_handles.rend();
+           ++iter) {
+        bool ok;
+        auto result =
+            ExtractComputationOpFromLastLivedVar(*iter, i, shrink_func, &ok);
```

Comment:
actually I am a little bit confused about ExtractComputationOpFromLastLivedVar function. I know it want to get the last op which will use this variable, but in reference count part, do we need to care about which op generate this variable?  What's the impact if we only care about the ops use this variable (the variable is used in the input part)

Reply:
`ExtractComputationOpFromLastLivedVar` returns the last op(s) which read(s) this variable or the last one op which writes this variables. See [HERE](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/details/reference_count_pass.cc#L154) in details.